client/my-sites/themes/theme-upload/index.jsx

```diff
@@ -161,6 +167,14 @@ class Upload extends React.Component {
 		const file = files[ 0 ] || files.item( 0 );
 		debug( 'zip file:', file );
 
+		if ( file.size > MAX_UPLOADED_THEME_SIZE ) {
+			notices.error(
+				translate( 'Theme zip is too big. Max theme size is 50 MB.' )
```

Comment:
I haven't read the whole code but I suppose that's for Jetpack sites or after the site was transferred (and thus made a JP site). There is only one error if the AT initiate endpoint fails and it is `initiate_failure`: https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/actions.js#L657-L657.